Abc.MoqComplete/Abc.MoqComplete/Services/AutoMockerMockedMethodProvider.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using JetBrains.ProjectModel;
+using JetBrains.ReSharper.Psi;
+using JetBrains.ReSharper.Psi.CSharp;
+using JetBrains.ReSharper.Psi.CSharp.Tree;
+using JetBrains.ReSharper.Psi.Resolve;
+using JetBrains.ReSharper.Psi.Tree;
+
+namespace Abc.MoqComplete.Services
+{
+	[SolutionComponent]
+	public class AutoMockerMockedMethodProvider : IAutoMockerMockedMethodProvider
```

Comment:
What's the difference with MockedMethodProvider ? A generic class may do the trick, if needed

Reply:
I am extract `BaseMethodProvider` and add abstract method `IsSetupMethod`. PR will be updated later